### ISSUE TITLE
Disable DataVirt UI when Komodo server isn't present

### ISSUE THREAD

**Issue by heiko-braun** [syndesisio/syndesis#5475]
In certain environments the DataVirt backend may not or cannot be installed. The user interface needs to suppress the dataVirt parts, when the komodo server isn't available to address these situations.

**Comment by heiko-braun** [comment]
@blafond Can you take a look at this?

**Comment by paoloantinori** [comment]
@gashcrumb @riccardo-forina do we have any suggestion for this?

**Comment by riccardo-forina** [comment]
If this info can be provided a build time, we could think of a `REACT_APP_ENABLE_DV` env variable to enable/disable the DV section. Not that the `REACT_APP_` prefix is required.

**Comment by gashcrumb** [comment]
Yep, other option is to add a feature toggle variable into the `config.json` file, that's another way we've done optional things like the activity logging etc.  Much clunkier than the environment variable though, and you have to be mindful that there's like several copies of it that get copied and used for various use-cases and the template generator needs to be updated if we worry about that still.

**Comment by zregvart** [comment]
I don't think we can do it at build time, enabling of Data virtualization is a install time thing.

**Comment by rareddy** [comment]
I was thinking to see if the backend server is available then show the "data" tab at runtime, but I guess that would ideal but little heavy handed. Is masthead completely cached? If it is then it may not be that bad to do it at runtime.

**Comment by heiko-braun** [comment]
What prevents usd doing at runtime using a preflight check as described by @rareddy ?

**Comment by riccardo-forina** [comment]
We can make the navigation dynamic and check for the DV presence at run-time, I think. 

I'd do this in a fire-and-forget way to avoid adding any boot lag to the app.
So the app would start with all the links visible in the sidebar but DV, while in the background a check against some API is performed and, if successful, the sidebar will be updated to show the new entry. 

Downside: the sidebar will noticeably change every time the app is loaded in a new tab/window (not on page transition) in a DV enabled install.

**Comment by zregvart** [comment]
Couldn't we simply check against a feature toggle in `config.json`, operator could update that value and for templated install we need to document how to change it along with how to scale the Komodo server?

**Comment by riccardo-forina** [comment]
Oh yeah the config.json is a better idea :D

**Comment by heiko-braun** [comment]
```
[...]
"features" : {
    "logging": false,
    "data-virt":false
  }
```

**Comment by asmigala** [comment]
Not sure this fits in how we want to present the DV option to users, but could we keep the Data entry in the sidebar, check that komodo is running when a user clicks on it, an in case it isn't, present a message saying that you need to start komodo in order to use DV?

**Comment by rareddy** [comment]
Can someone from fuse team help with this? Barry on PTO for next week.

**Comment by rareddy** [comment]
@rkorytkowski if you have some time do you want to take look at this issue?

**Comment by rkorytkowski** [comment]
@rareddy, sure, I can help with that tomorrow.

**Comment by heiko-braun** [comment]
@rkorytkowski Any idea how this can be addressed?

**Comment by rkorytkowski** [comment]
I'm implementing the idea of using config.json and showing a message after opening the Data link that Komodo needs to be setup if it is not available.

**Comment by rkorytkowski** [comment]
I'm uncertain which docs to update to describe enabling DataVirt UI, any pointers?

**Comment by zregvart** [comment]
> I'm uncertain which docs to update to describe enabling DataVirt UI, any pointers?

To be honest I'm not sure we have any. Perhaps [here](https://github.com/syndesisio/syndesis/blob/master/doc/sdh/cli/cmd_install.adoc).

I'm not sure where we document these cross project issues, perhaps @TovaCohen or @rareddy know?

**Comment by heiko-braun** [comment]
@rkorytkowski You mean "enabling" during the build?

@paoloantinori @blafond Do you know?

**Comment by rareddy** [comment]
Data Virtualization does not have any documents for syndesis/fuse online yet. As per where it needs to be, I think it depends on what @heiko-braun is asking. Is this a dynamic thing vs installation time? If installation time then I agree with the above link for installation instructions. If it is installation, it would be better if we tie with the same flag how the backend is controlled so that we have a single way to open up UI/Server.

**Comment by rkorytkowski** [comment]
@heiko-braun, it can be enabled during installation. @rareddy, what's the flag for the backend?

**Comment by rareddy** [comment]
see the issues #4203 #5435

**Comment by TovaCohen** [comment]
Robert (Bob) Roldan is the writer for any user doc that we plan to provide for Data Virt. I'll ping him.

**Comment by rkorytkowski** [comment]
In my PR I've introduced --datavirt to the syndesis install scripts, which enables Data Virtualizations.

Also I used config.json:datavirt.enabled instead of config.json:features.datavirt since I wanted to reuse DATAVIRT_ENABLED template setting defined for backend, which is either 0 or 1. It fits better under the datavirt section since features use false or true values and it might have been confusing.

**Comment by zregvart** [comment]
@rkorytkowski I think that's fine for the first step, when we transition from templates to pure operator based install we can correct this.

**Comment by heiko-braun** [comment]
@rkorytkowski I am reopening this due to #6015

**Comment by rkorytkowski** [comment]
Addressed in #6015. The issue was on master only.

**Comment by asmigala** [comment]
Verified with 1.7.12
